src/main/java/com/bravo/user/model/filter/PaymentFilter.java

```diff
@@ -0,0 +1,17 @@
+package com.bravo.user.model.filter;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.util.Set;
+
+@Data
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
```

Comment:
AllArgsConstructor was needed in order to create the object in the retrievePaymentByUserId on line 39 in the PaymentService class. 

NoArgsConstructor - needed to stop an issue with serialization of the filter request object, it would return a Bad Request due to the application not being able to deserialize the object. Error from terminal: (although at least one Creator exists): cannot deserialize from Object value (no delegate- or property-based Creator).

I could get rid of the AllArgsConstructor by initializing an instance of the 
object and then just using a setter, but the code does look cleaner this way.

Reply:
Code generators like Lombok are a toss up.  In this case, it's autogen needs to match the serialization.  My preference has evolved to using immutable builders in this situation and configuring the serialization to use builders.  The main idea here is to pick one way to set the values and stick to it.  Someone copied a few pages from effective java here https://cs108.epfl.ch/archive/19/c/i/EffectiveJava_Item17.pdf.  I highly recommend the book.  It helped my view of Java coding immensely.  The goal should be immutable value objects.

Reply:
Thank you for the share, I will read this and take a look at getting the book (always room for learning more).